frontend/hub/collections/hooks/useCopyToRepository.tsx

```diff
@@ -0,0 +1,252 @@
+        } &&version=${collection.collection_version?.version || ''}`
+      );
+
+      if (repos.data?.length > 0) {
+        const reposMapped = repos.data.map((item) => item.repository || ({} as Repository));
+        setFixedRepositories(reposMapped);
+      }
+    }
+
+    void (async () => {
+      await getSelected();
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  let queryParams = undefined;
+  if (operation == 'approve') {
+    queryParams = { pulp_label_select: 'pipeline=approved' };
+  }
+
+  const view = usePulpView({
+    url: pulpAPI`/repositories/ansible/ansible/`,
+    toolbarFilters,
+    tableColumns,
+    disableQueryString: true,
+    keyFn: (item) => item.name,
+    queryParams,
+  });
+
+  return (
+    <Modal
+      title={t(`Select repositories`)}
+      aria-label={t(`Select repositories`)}
+      isOpen
+      onClose={() => {
+        props.onClose();
+      }}
+      variant={ModalVariant.large}
+      tabIndex={0}
+      actions={[
+        <Button
+          key="select"
+          variant="primary"
+          id="select"
+          onClick={() => {
+            void (async () => {
+              await copyToRepositories();
+            })();
+          }}
+          isDisabled={selectedRepositories.length == 0}
+          isLoading={isLoading}
+        >
+          {t('Select')}
+        </Button>,
+        <Button
+          key="cancel"
+          variant="link"
+          onClick={() => {
+            props.onClose();
+          }}
+        >
+          {t('Cancel')}
+        </Button>,
+      ]}
+      hasNoBodyWrapper
+    >
+      <PageTable<AnsibleAnsibleRepositoryResponse>
+        id="hub-copy-to-repository-table"
+        toolbarFilters={toolbarFilters}
+        tableColumns={tableColumns}
+        errorStateTitle={t('Error loading collections')}
+        emptyStateTitle={t('No collections yet')}
+        emptyStateDescription={t('To get started, upload a collection.')}
+        emptyStateButtonText={t('Upload collection')}
+        {...view}
+        defaultTableView="list"
+        defaultSubtitle={t('Repository')}
+        showSelect={true}
+        compact={true}
+        selectedItems={selectedRepositories as AnsibleAnsibleRepositoryResponse[]}
+        isSelectMultiple={true}
+        isSelected={(item) =>
+          selectedRepositories.find((i) => i.name == item.name) ||
+          fixedRepositories.find((i) => i.name == item.name)
+            ? true
+            : false
+        }
+        selectItem={(item) => {
+          const newItems = [...selectedRepositories, item];
+          setSelectedRepositories(newItems);
+        }}
+        selectItems={(items) => {
+          const newItems = [...selectedRepositories];
+          for (const item of items) {
+            if (
+              !selectedRepositories.find((item2) => item.name == item2.name) &&
+              !fixedRepositories.find((item2) => item2.name != item.name)
```

Comment:
```suggestion
              !fixedRepositories.find((item2) => item2.name == item.name)
```

Reply:
Ah, thats hidden error that was not even relevant :D Because selected repos are displayed based on both selected and fixed and if selected contains fixed repos (repos that collection is already in), it does not matter, because API will ignore that.